examples/dependency/README.md

```diff
@@ -155,6 +155,24 @@ After a while, assuming the controller of this CRD will reconcile and make the `
 Let's update it manually. 
 
 ```shell script
+$ kubectl  edit components source -o yaml
+apiVersion: core.oam.dev/v1alpha2
+kind: Component
+metadata:
+  name: source
+spec:
+  workload:
+    apiVersion: example.com/v1
+    kind: Foo
+    metadata:
+      name: source
+    spec:
+      key: test-value
+-   status:
+-      state: pending
++ #   status:
++ #      state: pending
+
```

Comment:
why edit component ?

Reply:
this component yaml set `.status.state` to `pending`.

if we only edit `source` workload's `.status.state` and set it to `running`,  then it will be reconciled to `pending` state according to component's value soon.

Reply:
I can't very understand here. According to the context, the `status.state` is [already pending](https://github.com/crossplane/oam-kubernetes-runtime/pull/142/files#diff-c4f5f305c3bca0c5fea1e4f20a8d5ad2R137-R138) , and in your example, your are removing the field?

Reply:
`.status.state` is set to `pending` in  Component `source`. 
https://github.com/crossplane/oam-kubernetes-runtime/blob/55cd9dbe93cfeba4fe03dfd8c81c34a4f367d2bc/examples/dependency/demo-with-conditions.yaml#L7-L15

Even we modify `.status.state` of workload Foo `source` and set the value to `running`, it will be re-rendered to `pending` according to  Component `source`'s definition(by application controller). 

i will re-experiment this use case  to check it.

Reply:
I execute this case again, yes, editing Foo `source`'s `state` is enough, and the Foo `sink` object will be create after the modification. but Foo `source`'s `state`  will also be reconciled back to `pending` because of Component `source`'s `state` value. the key is which come first.  

i will remove the this  component editing part.